### PR TITLE
Fix create react apps by adding node related files to default local

### LINKED ISSUE #2074: Support and test applications created with "Create React App"
https://create-react-app.dev/

Make sure running applications generated with create-react-app are supported out of the box with the default settings and do not require any special configuration.

For that we need to add some stuff to the default local paths, see relevant logs:

```
2023-11-14T15:47:41.420568Z TRACE ThreadId(01) start:handle_client_message:respond: mirrord_agent: 
  new cancellation_token=CancellationToken { is_cancelled: false } 
  FileRequest(Open(OpenFileRequest { path: "/.yarnrc", open_options: OpenOptionsInternal { read: true, write: false, append: false, truncate: false, create: false, create_new: false } })) 
  response=File(Open(Err(RemoteIO(RemoteIOError { raw_os_error: Some(2), kind: NotFound }))))
  
2023-11-14T15:47:41.427034Z TRACE ThreadId(01) start:handle_client_message:respond: mirrord_agent: 
  new cancellation_token=CancellationToken { is_cancelled: false } 
  FileRequest(Open(OpenFileRequest { path: "/.yarnrc.yml", open_options: OpenOptionsInternal { read: true, write: false, append: false, truncate: false, create: false, create_new: false } })) 
  response=File(Open(Err(RemoteIO(RemoteIOError { raw_os_error: Some(2), kind: NotFound }))))
  
2023-11-14T15:47:41.777375Z TRACE ThreadId(01) start:handle_client_message:respond: mirrord_agent: 
  new cancellation_token=CancellationToken { is_cancelled: false } 
  FileRequest(Access(AccessFileRequest { pathname: "/yarn.json", mode: 0 })) 
  response=File(Access(Err(RemoteIO(RemoteIOError { raw_os_error: Some(2), kind: NotFound }))))
  
2023-11-14T15:47:42.437801Z TRACE ThreadId(01) start:handle_client_message:respond: mirrord_agent: 
  new cancellation_token=CancellationToken { is_cancelled: false } 
  FileRequest(Access(AccessFileRequest { pathname: "/node_modules", mode: 0 })) 
  response=File(Access(Err(RemoteIO(RemoteIOError { raw_os_error: Some(2), kind: NotFound }))))
```
source: https://canary.discord.com/channels/933706914808889356/1173653041791123716/1174023486495264789

Also add a test with such an app to prevent regressions.

### PR DESCRIPTION
**Pull request by aviramha**
Closes #2074

### CODE REVIEW

File: mirrord/layer/src/file/filter/read_local_by_default.rs [hunk 1]
```diff
@@ -46,6 +46,11 @@ pub fn regex_set_builder() -> RegexSetBuilder {
         // TODO: `node` searches for this file in multiple directories, bypassing some of our
         // ignore regexes, maybe other "project runners" will do the same.
         r"^.*/package.json$",
+        r"^.*/.yarnrc$",
+        r"^.*/.yarnrc.yml$",
+        r"^.*/.yarnrc.json$",
+        r"^.*/node_modules",
+        r"^.*/package.json$",
```
Comment (t4lz) [inline mirrord/layer/src/file/filter/read_local_by_default.rs:53]:
```suggestion
```

already present above

File: mirrord/layer/src/file/filter/read_local_by_default.rs [hunk 2]
```diff
@@ -46,6 +46,11 @@ pub fn regex_set_builder() -> RegexSetBuilder {
         // TODO: `node` searches for this file in multiple directories, bypassing some of our
         // ignore regexes, maybe other "project runners" will do the same.
         r"^.*/package.json$",
+        r"^.*/.yarnrc$",
+        r"^.*/.yarnrc.yml$",
+        r"^.*/.yarnrc.json$",
+        r"^.*/node_modules",
```
Comment (t4lz) [inline mirrord/layer/src/file/filter/read_local_by_default.rs:52]: Nitpick: `^.*` is superfluous. We can leave it there thought if you think it's more readable or something.

```suggestion
        r"/\.yarnrc$",
        r"/\.yarnrc.yml$",
        r"/\.yarnrc.json$",
        r"/node_modules",
```

**Review by t4lz — APPROVED** [review]

File: mirrord/layer/src/file/filter/read_local_by_default.rs (hunk 2 above)
Comment (t4lz) [reply mirrord/layer/src/file/filter/read_local_by_default.rs:52]: And is it on purpose that the `node_modules` one does not end with a `$`?

Comment (aviramha) [reply mirrord/layer/src/file/filter/read_local_by_default.rs:52]: yeah I want to get it on any path

File: mirrord/layer/src/file/filter/read_local_by_default.rs [hunk 3]
```diff
@@ -46,6 +46,10 @@ pub fn regex_set_builder() -> RegexSetBuilder {
         // TODO: `node` searches for this file in multiple directories, bypassing some of our
         // ignore regexes, maybe other "project runners" will do the same.
         r"^.*/package.json$",
+        r"/\.yarnrc$",
+        r"/\.yarnrc.yml$",
+        r"/\.yarnrc.json$",
+        r"/node_modules$",
```
Comment (t4lz) [inline mirrord/layer/src/file/filter/read_local_by_default.rs:52]:
```suggestion
        r"/node_modules",
```
Ooops, sorry

**Pull request merged**